go/chat/boxer.go

```diff
@@ -77,13 +83,27 @@ func (b *Boxer) makeErrorMessage(msg chat1.MessageBoxed, err UnboxingError) chat
 	})
 }
 
-// UnboxMessage unboxes a chat1.MessageBoxed into a keybase1.Message.  It finds
-// the appropriate keybase1.CryptKey.
-// The first return value is unusable if the err != nil
-// Returns (_, err) for non-permanent errors, and (MessageUnboxedError, nil) for permanent errors.
-// Permanent errors can be cached and must be treated as a value to deal with.
-// Whereas temporary errors are transient failures.
-func (b *Boxer) UnboxMessage(ctx context.Context, boxed chat1.MessageBoxed, finalizeInfo *chat1.ConversationFinalizeInfo) (chat1.MessageUnboxed, UnboxingError) {
+// UnboxMessage unboxes a chat1.MessageBoxed into a keybase1.Message. It finds
+// the appropriate keybase1.CryptKey, decrypts the message, and verifies
+// several things:
+//   - The message's signature is valid.
+//   - (TODO) The signing KID was valid when the signature was made.
+//   - (TODO) The signing KID belongs to the sending device.
+//   - (TODO) The sending device belongs to the sender.
+//     [Note that we do currently check the KID -> UID relationship,
+//     independent of the device ID.]
+//   - (TODO) The sender has write permission in the TLF.
+//   - (TODO) The TLF name, public flag, and finalized info resolve to the TLF ID.
+//   - The conversation ID derives from the ConversationIDTriple.
+//   - The body hash is not a replay from another message we know about.
+//   - The prev pointers are consistent with other messages we know about.
+//   - (TODO) The prev pointers are not absurdly ancient.
+//
+// The first return value is unusable if the err != nil Returns (_, err) for
+// non-permanent errors, and (MessageUnboxedError, nil) for permanent errors.
+// Permanent errors can be cached and must be treated as a value to deal with,
+// whereas temporary errors are transient failures.
+func (b *Boxer) UnboxMessage(ctx context.Context, boxed chat1.MessageBoxed, convID chat1.ConversationID, finalizeInfo *chat1.ConversationFinalizeInfo) (chat1.MessageUnboxed, UnboxingError) {
```

Comment:
Now there's 3 convID-type things
- boxed.ClientHeader.Conv
- unboxed.ClientHeader.Conv
- the convID argument

This now asserts the last two are the same. Maybe we should assert the first one in as well.

That's part of a family of things that should be but are not yet asserted to be the same between the signed header and the unsigned header. As you're listing things to assert it might be worth throwing in something like
`"- (TODO) make sure the signed and unsigned header agree on as much as possible"`. I still don't know what we're using the unsigned header for in the client.